variant/validators/genomic_deletion.py

```diff
@@ -0,0 +1,180 @@
+"""The module for Genomic Deletion Validation."""
+from variant.validators.deletion_base import DeletionBase
+from variant.schemas.classification_response_schema import \
+    ClassificationType
+from variant.schemas.token_response_schema import GenomicDeletionToken
+from .genomic_base import GenomicBase
+from typing import List
+from variant.schemas.classification_response_schema import Classification
+from variant.schemas.token_response_schema import GeneMatchToken
+from variant.schemas.validation_response_schema import ValidationResult
+from variant.schemas.token_response_schema import Token
+import logging
+
+
+logger = logging.getLogger('variant')
+logger.setLevel(logging.DEBUG)
+
+
+class GenomicDeletion(DeletionBase):
+    """The Genomic Deletion Validator class."""
+
+    def validate(self, classification: Classification) \
+            -> List[ValidationResult]:
+        """Validate a given classification.
+
+        :param Classification classification: A classification for a list of
+            tokens
+        :return: A list of validation results
+        """
+        results = list()
+        errors = list()
+
+        classification_tokens = self.get_classification_tokens(classification)
+        gene_tokens = self.get_gene_tokens(classification)
+
+        if gene_tokens and len(gene_tokens) > 1:
+            errors.append('More than one gene symbol found for a single'
+                          f' {self.variant_name()}')
+
+        if len(classification.non_matching_tokens) > 0:
+            errors.append(f"Non matching tokens found for "
+                          f"{self.variant_name()}.")
+
+        genomic_base = GenomicBase(self.dp)
+        nc_accessions = genomic_base.get_nc_accessions(classification)
+        if not nc_accessions:
+            errors.append('Could not find NC_ accession for '
+                          f'{self.variant_name()}')
+
+        if len(errors) > 0:
+            return [self.get_validation_result(
+                classification, False, 0, None,
+                '', '', errors, gene_tokens)]
+
+        self.get_valid_invalid_results(classification_tokens, nc_accessions,
+                                       classification, results, gene_tokens)
+        return results
+
+    def get_hgvs_expr(self, classification, t, s, is_hgvs) -> tuple:
+        """Return HGVS expression and whether or not it's an Ensembl transcript
+
+        :param Classification classification: A classification for a list of
+            tokens
+        :param str t: Transcript retrieved from transcript mapping
+        :param bool is_hgvs: Whether or not classification is HGVS token
+        :return: A tuple containing the hgvs expression and whether or not
+            it's an Ensembl Transcript
+        """
+        if t.startswith('ENST'):
+            # TODO
```

Comment:
this is #91, right?

Reply:
I can't exactly remember what the todo is for. I know I need to go back and look at ENST because they cause some trouble sometimes.